ferrocene/doc/internal-procedures/src/manual-upstream-pulls.rst

```diff
@@ -10,97 +10,70 @@ What is an Upstream Pull
-  if doing a `master` upstream pull and `beta` for a `beta` upstream pull.
-* `<NEW_PULL_BRANCH>`: This represents the branch you will be making the pull
-  with. Replace it with a branch name of your choice, e.g.
-  `pull-upstream-master-2023-05-22`
+Upstream Pull with merge conflicts
+----------------------------------
 
-Initiating the pull
-^^^^^^^^^^^^^^^^^^^
+If an automatic upstream pull creates merge conflicts, the automation commits the
+merge conflicts and they need to be resolved manually.
 
-Run the following git commands to set up a new local `<NEW_PULL_BRANCH>`
-branch to do the pull on::
+Checkout PR
+^^^^^^^^^^^
 
-  git checkout <BASE_BRANCH> # check out the ferrocene branch to do the pull for
-  git pull --rebase # make sure your local branch is up to date
-  git submodule update # update any submodules that might've changed
-  git checkout -b <NEW_PULL_BRANCH> # checkout a new branch to do the pull with
+Start by checking out the PR. The name of the branch is
+`automation/pull-upstream/<ID>`, e.g. `automation/pull-upstream/dfvjj2s4`::
 
-Performing the pull
-^^^^^^^^^^^^^^^^^^^
+    git switch automation/pull-upstream/<ID>
 
-Now run::
+Fix conflict
+^^^^^^^^^^^^
 
-  ferrocene/tools/pull-upstream/pull.sh <UPSTREAM_BRANCH>
+There are multiple kinds of merge conflicts:
 
-The script will fetch the latest changes from `rust-lang/rust`'s
-`<UPSTREAM_BRANCH>` branch, create a commit to remove the excluded files
-(see below) from the upstream branch, and finally try to merge upstream into
-the current branch.
+General conflict
+""""""""""""""""
 
-The script is able to resolve some kinds of merge conflicts on its own. If
-there are still remaining merge conflicts after the automatic resolution the
-script will exit. You'll have to fix the merge conflicts yourself in that case:
+In order to find general conflicts search for the git conflict markers (``<<<<<<<``,
+``=======`` or ``>>>>>>>``).
 
-There are two kinds of conflicts you will encounter in the output that the
-script produces:.
+Fix the conflict manually and stage the changed files::
 
-* Submodule conflict ::
+  git add <FILE_1> <FILE_2> # stages file(s)
 
-    Please manually handle the merging of each conflicted submodule.
-    This can be accomplished with the following steps:
-    - go to submodule (src/tools/cargo), and either merge commit <COMMIT_HASH>
-      or update to an existing commit which has merged those changes
-    - come back to superproject and run:
+Deletion conflict
+"""""""""""""""""
 
-          git add src/tools/cargo
+In order to find deletion conflicts search for ``<<<PULL-UPSTREAM>>>``. Automation
+inserts this marker into the first line of the conflicting file.
 
-      to record the above merge or update
-    - resolve any other conflicts in the superproject
-    - commit the resulting index in the superproject
+This conflict happens when we made changes to a file and it gets deleted by upstream.
+To resolve this, remove the file::
```

Comment:
Not sure exactly how to word this, but we shouldn't blindly delete the files (otherwise we could have the automation do it for you), but also verify that we don't remove load-bearing changes we made.

For example, if we made a change in a bootstrap file managed by upstream, and upstream removes the file, if we were to just remove the file we'd lose the change we made.

Reply:
That's what I try to communicate with the sections below ("If this file was a test case ..."). I will word it more generally.